docs/install/airgapped-clusters.md

```diff
@@ -0,0 +1,365 @@
+
+> Edit the variables `REGISTRY_IMG_ETCD_OPERATOR` and `REGISTRY_IMG_ETCD_PROXY`
+
+```
+# Change the 2 images on the etcd-operator.yaml for your registry URL
+ETCD_OPERATOR_VERSION=develop
+
+# Etcd operator controller image
+REGISTRY_IMG_ETCD_OPERATOR=my-registry-url/storageos/etcd-cluster-operator-controller:$ETCD_OPERATOR_VERSION
+sed -i -e "s#image: storageos/etcd-cluster-operator-controller:$ETCD_OPERATOR_VERSION#image: $REGISTRY_IMG_ETCD_OPERATOR#g" etcd-operator.yaml
+
+REGISTRY_IMG_ETCD_PROXY=my-registry-url/storageos/etcd-cluster-operator-proxy:$ETCD_OPERATOR_VERSION
+sed -i -e "s#image: storageos/etcd-cluster-operator-proxy:$ETCD_OPERATOR_VERSION#image: $REGISTRY_IMG_ETCD_PROXY#g" etcd-operator.yaml
+
+# check the images are set correctly
+grep -C2 "image:" etcd-operator.yaml
+```
+
+3. Once the images are set on the `etcd-operator.yaml` it is required to add the
+reqistry URL as an argument to the Etcd operator.
+
+Add the flag `-
+--etcd-repository=$REGISTRY/quay.io/coreos/etcd`  in
+the args of the `manager` container defined in `etcd-operator.yaml`
+
+4. Edit the variabled $REGISTRY
+
+```
+REGISTRY=my-registry-url
+
+# Old 
+      - args:
+        - --enable-leader-election
+        - --proxy-url=storageos-proxy.storageos-etcd.svc
+        command:
+        - /manager
+
+# New
+      - args:
+        - --enable-leader-election
+        - --proxy-url=storageos-proxy.storageos-etcd.svc
+        - --etcd-repository=$REGISTRY/quay.io/coreos/etcd # Edit this line with your registry url
+        command:
+        - /manager
+```
+
+5. Create the Etcd cluster definition, `etcd-cluster.yaml`.
+
+```yaml
+apiVersion: etcd.improbable.io/v1alpha1
+kind: EtcdCluster
+metadata:
+  name: storageos-etcd
+spec:
+  version: 3.5.0
+  podTemplate:
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - podAffinityTerm:
+            labelSelector:
+              matchExpressions:
+              - key: etcd.improbable.io/cluster-name
+                operator: In
+                values:
+                - storageos-etcd
+            topologyKey: kubernetes.io/hostname
+          weight: 100
+    etcdEnv:
+    - name: ETCD_HEARTBEAT_INTERVAL
+      value: "500"
+    - name: ETCD_ELECTION_TIMEOUT
+      value: "5000"
+    - name: ETCD_MAX_SNAPSHOTS
+      value: "10"
+    - name: ETCD_MAX_WALS
+      value: "10"
+    - name: ETCD_QUOTA_BACKEND_BYTES
+      value: "8589934592"
+    - name: ETCD_SNAPSHOT_COUNT
+      value: "100000"
+    - name: ETCD_AUTO_COMPACTION_RETENTION
+      value: "20000"
+    - name: ETCD_AUTO_COMPACTION_MODE
+      value: revision
+    resources:
+      limits:
+        cpu: 200m
+        memory: 200Mi
+      requests:
+        cpu: 200m
+        memory: 200Mi
+  replicas: 3
+  storage:
+    volumeClaimTemplate:
+      resources:
+        requests:
+          storage: 128Gi
+      storageClassName: local-path
+  tls:
+    enabled: false
```

Comment:
By default or there's no current functionality in the operator?

Reply:
I noticed that the plugin allows for etcd with TLS via https://github.com/ondat/documentation/pull/36/files#diff-c9b44c77abf967dd8ddd90e8b5f64ee7fbc810c19f37593d1f79c53ed10c1af6R47, hence the question - I wonder if it's worth testing to see if we can make it tls-enabled safely as a best practice, and also updating our other documentation?